### ISSUE TITLE
Wrong return type in AMQP Transport Connection with newest version of php-amqp extension

### ISSUE THREAD

**Issue by texnicii** [symfony/symfony#51282]
### Symfony version(s) affected

5.4.0

### Description

Since [v2.0.0alpha1](https://github.com/php-amqp/php-amqp/releases/tag/v2.0.0alpha1) of php-amqp extension the method `AMQPQueue::ack()` returned `void` instead `bool`

After updating version for php-amqp extension I am getting type error at `Symfony\Component\Messenger\Bridge\Amqp\Transport\Connection::ack()` which using `AMQPQueue::ack()`

### How to reproduce

1. Update php-amqp extension to [v2.0.0alpha1](https://github.com/php-amqp/php-amqp/tree/v2.0.0alpha1) or [latest](https://github.com/php-amqp/php-amqp/tree/latest) version
2. Run the symfony messenger worker `./bin/console messenger:consume routing_name`
3. Dispatch some message in order to consumer will receive it and you will get the type error in messenger worker

### Possible Solution

I guess need to change the return type to `void` in `Symfony\Component\Messenger\Bridge\Amqp\Transport\Connection::ack()` because no one using this returned type.

Changes may looks like this
```php
public function ack(\AMQPEnvelope $message, string $queueName): void
    {
        $this->queue($queueName)->ack($message->getDeliveryTag());
    }
```

Because `Symfony\Component\Messenger\Bridge\Amqp\Transport\AmqpReceiver` not using this returned value
```php
public function ack(Envelope $envelope): void
    {
        try {
            $stamp = $this->findAmqpStamp($envelope);

            $this->connection->ack(
                $stamp->getAmqpEnvelope(),
                $stamp->getQueueName()
            );
        } catch (\AMQPException $exception) {
            throw new TransportException($exception->getMessage(), 0, $exception);
        }
    }
```

**Comment by fabpot** [comment]
Cc @lyrixx

**Comment by stefangr** [comment]
Should be fixed in #51168

**Comment by xabbuh** [comment]
let's close here then, as the changes from #51168 will be part of the next patch release
